main.tf

```diff
@@ -38,16 +38,216 @@ resource "aws_lb" "main" {
+          issuer                              = authenticate_oidc.value["issuer"]
+          token_endpoint                      = authenticate_oidc.value["token_endpoint"]
+          user_info_endpoint                  = authenticate_oidc.value["user_info_endpoint"]
+          on_unauthenticated_request          = lookup(authenticate_oidc.value, "on_unauthenticated_request", null)
+          scope                               = lookup(authenticate_oidc.value, "scope", null)
+          session_cookie_name                 = lookup(authenticate_oidc.value, "session_cookie_name", null)
+          session_timeout                     = lookup(authenticate_oidc.value, "session_timeout", null)
+        }
+      }
+
+      dynamic "fixed_response" {
+        for_each = length(keys(lookup(default_action.value, "fixed_response", {}))) == 0 ? [] : [lookup(default_action.value, "fixed_response", {})]
+
+        content {
+          content_type = fixed_response.value["content_type"]
+          message_body = lookup(fixed_response.value, "message_body", null)
+          status_code  = lookup(fixed_response.value, "status_code", null)
+        }
+      }
+
+      dynamic "redirect" {
+        for_each = length(keys(lookup(default_action.value, "redirect", {}))) == 0 ? [] : [lookup(default_action.value, "redirect", {})]
+
+        content {
+          path        = lookup(redirect.value, "path", null)
+          host        = lookup(redirect.value, "host", null)
+          port        = lookup(redirect.value, "port", null)
+          protocol    = lookup(redirect.value, "protocol", null)
+          query       = lookup(redirect.value, "query", null)
+          status_code = redirect.value["status_code"]
+        }
+      }
+
+      ## For HTTPS listeners only
+      dynamic "authenticate_cognito" {
+        for_each = length(keys(lookup(default_action.value, "authenticate_cognito", {}))) == 0 ? [] : [lookup(default_action.value, "authenticate_cognito", {})]
+
+        content {
+          # Max 10 extra params
+          authentication_request_extra_params = lookup(authenticate_cognito.value, "authentication_request_extra_params", null)
+          on_unauthenticated_request          = lookup(authenticate_cognito.value, "on_authenticated_request", null)
+          scope                               = lookup(authenticate_cognito.value, "scope", null)
+          session_cookie_name                 = lookup(authenticate_cognito.value, "session_cookie_name", null)
+          session_timeout                     = lookup(authenticate_cognito.value, "session_timeout", null)
+          user_pool_arn                       = authenticate_cognito.value["user_pool_arn"]
+          user_pool_client_id                 = authenticate_cognito.value["user_pool_client_id"]
+          user_pool_domain                    = authenticate_cognito.value["user_pool_domain"]
+        }
+      }
+    }
+  }
+}
```

Comment:
Add a note in changelog to add authentication, redirect and fixed response in unreleased.
Had initially purposed to restructure this block together with target group block the same way we have the current S3 module, kindly see if it is possible